torchci/lib/bot/pytorchBotHandler.ts

```diff
@@ -157,46 +222,55 @@ The explanation needs to be clear on why this is needed. Here are some good exam
 
     if (forceRequested) {
       rejection_reason = await this.reasonToRejectForceRequest(forceMessage);
+    } else {
+      // Ensure the PR has been signed off on
+      let isApprovedPR = await this.doesHaveApprovedStatus()
+      if (!isApprovedPR) {
+        rejection_reason = "This PR needs to be approved by an authorized maintainer before merge."
+      }
+    }
+
+    if (rejection_reason) {
+      await this.logger.log("merge-error", extra_data);
```

Comment:
note: from this point down in this file it's just a minor refactor of inverting the old `if (!rejection_reason)` block